script/unicode/Unicode/UnicodeSkipList.lean

```diff
@@ -0,0 +1,148 @@
+/-
+Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jean-Baptiste Tristan
+-/
+import Init.Data.Char.UnicodeSkipList
+import Unicode.Unicode
+import Unicode.Parse
+import Init.Data.Char.Basic
+
+open Std System IO FilePath FS
+
+namespace Char.UnicodeSkipList
+
+/-
+The following code creates the unicode skip list data structure for
+a given property. To understand it at a high-level, we provide an
+explanation in Init/Data/Char/UnicodeSkipList.lean
+-/
+
+/-
+Break down the sequence of codepoints into a sequence of ranges,
+alternating between ranges that satisfy the property and ranges that
+do not.
+-/
+def explicitRanges (ucd : List UnicodeData) (property : UnicodeData → Bool) : List Range := Id.run do
+  let mut rangeOpt : Option Range := none
+  let mut ranges := []
```

Comment:
I think it's quite confusing that a range in `rangeOpt` is a closed interval while a range in `ranges` is a half-open interval. `Std.Range` is always meant to be a half-open interval, so I think it would be good if your code used it that way.

Reply:
They are ranges under construction.

Reply:
In that code, the ranges are under construction, so they are half-open with respect to the previous iteration that finished, and closed with respect to the current one. Perhaps I should not use ranges has an intermediate data structure, but I think the resulting code would be more complicated. I'd rather not change it, but it is your call.